src/storage/raw.c

```diff
@@ -90,18 +103,23 @@ append(struct Storage* self_, const struct VideoFrame* frames, size_t* nbytes)
     return DeviceState_Running;
 Error:
     *nbytes = 0;
-    return stop(self_);
+    return raw_stop(self_);
 }
 
 static void
-destroy(struct Storage* writer_)
+raw_destroy(struct Storage* writer_)
 {
     struct Raw* self = containerof(writer_, struct Raw, writer);
-    stop(writer_);
+    raw_stop(writer_);
     storage_properties_destroy(&self->properties);
     free(self);
 }
 
+static void
+raw_reserve_image_shape(struct Storage* self_, const struct ImageShape* shape)
+{
+}
```

Comment:
Consider commenting to indicate an empty body is expected
```suggestion
{ // no-op
}
```

An alternative to adding these empty functions is to allow the driver to set the function pointer to 0. This would require a change in the hal and a doc update in the kit.